src/java.base/share/classes/jdk/internal/foreign/Utils.java

```diff
@@ -56,6 +56,13 @@ public final class Utils {
 
     public static final boolean IS_WINDOWS = privilegedGetProperty("os.name").startsWith("Windows");
 
+    static {
```

Comment:
Another possibility here is to make sure that the static methods here (and also in `NativeMemorySegmentImpl` go via `reinterpret`, so that there's nothing additional to do.

Reply:
Yes, that's also an option, but `reintepret` is CallerSensitive, so this does introduce some overhead in terms of startup perf I think.

At the same time, it seems to require a similar amount of care to be taken, so I'm not sure how much it's an improvement over `ensureClassInitialized`.